pkg/tsdb/prometheus/prometheus.go

```diff
@@ -49,20 +49,28 @@ func newInstanceSettings(httpClientProvider httpclient.Provider, cfg *setting.Cf
 		}
 		roundTripper, err := httpClientProvider.GetTransport(*opts)
 		if err != nil {
-			return nil, fmt.Errorf("error creating http client: %v", err)
+			return nil, fmt.Errorf("error creating http round tripper: %v", err)
 		}
 
+		// Older version using standard Go Prometheus client
 		b, err := buffered.New(roundTripper, tracer, settings, plog)
 		if err != nil {
 			return nil, err
 		}
 
-		qd, err := querydata.New(httpClientProvider, cfg, features, tracer, settings, plog)
+		httpClient, err := httpClientProvider.New(*opts)
+		if err != nil {
+			return nil, fmt.Errorf("error creating http client: %v", err)
+		}
+
+		// New version using custom client and better response parsing
+		qd, err := querydata.New(httpClient, features, tracer, settings, plog)
```

Comment:
Because we reuse the httpClient and it's setup we don't need duplicated azure setup.